user.py

```diff
@@ -35,14 +35,21 @@ def _check_id(self, message: discord.Message) -> Optional[int]:
         try:
             # If ping is typed out by user using their ID, it doesn't count as a mention
             # Thus, try and match with regex
-            check_ping = re.search(r"<@!?(\d+)>", content)
+
+            # Match at least 4 integers to not confuse user ids with regular numbers
+            # User ids are snowflakes. Based on the snowflake format (https://discord.com/developers/docs/reference#snowflakes)
+            # Any account created after the first millisecond of 2015 will have an id of at least 4194304 (1 << 22)
+            # Which is 7 integers long, so this practically shouldn't ever miss anyone.
+            check_ping = re.search(r"<@!?(\d{4,})>", content)
```

Comment:
I'm not sure I understand the purpose of this. Are there circumstances where we would get the pattern `<@123>` with only 3 or fewer digits?

Reply:
Good catch, this part doesn't need to be modified to fix the issue with `$edit` - I misread it.

Thanks! Undid these changes.